ui/src/components/Timeline/Timeline.tsx

```diff
@@ -64,13 +64,9 @@ const Timeline: FC<TimelineProps> = ({
   const sortedEntities = selectSortedEntities(state);
 
   const items = sortedEntities.map(
-    (entity) => new TimelineItem(entity, layout)
+    (entity) => new TimelineItem(entity, state.layout)
   );
 
-  useEffect(() => {
-    onLayoutUpdate(state.layout);
-  }, [state.layout, onLayoutUpdate]);
```

Comment:
This was the problematic section as the `onLayoutUpdate` callback was called not only when the layout had actually changed, but also on initial render.